yorkie/src/test/kotlin/dev/yorkie/document/json/JsonTreeTest.kt

```diff
@@ -299,6 +315,142 @@ class JsonTreeTest {
+                    listOf(0, 0),
+                    listOf(0, 0),
+                    TreeNode("text", value = "X"),
+                    "$.t",
+                ),
+                TreeStyleOpInfo(
+                    0,
+                    0,
+                    listOf(0, -7), // TODO: need to check if these are the actual right values???
+                    listOf(0, -7),
+                    mapOf("a" to "b"),
+                    "$.t",
+                ),
+            ),
+            actualOperationDeferred.await(),
+        )
+    }
+
+    @Test
+    fun `should emit Tree edit operations accordingly when edited with path`() = runTest {
+        val document = Document(Document.Key(""))
+        fun JsonObject.tree() = getAs<JsonTree>("t")
+
+        document.updateAsync {
+            it.setNewTree(
+                "t",
+                ElementNode(
+                    "doc",
+                    children = listOf(
+                        ElementNode(
+                            "tc",
+                            children = listOf(
+                                ElementNode(
+                                    "p",
+                                    children = listOf(
+                                        ElementNode(
+                                            "tn",
+                                            children = listOf(TextNode("ab")),
+                                        ),
+                                    ),
+                                ),
+                            ),
+                        ),
+                    ),
+                ),
+            )
+        }.await()
+        assertEquals(
+            """<doc><tc><p><tn>ab</tn></p></tc></doc>""",
+            document.getRoot().tree().toXml(),
+        )
+
+        val actualOperationDeferred = async(UnconfinedTestDispatcher()) {
+            document.events("$.t")
+                .filterIsInstance<LocalChange>()
+                .flatMapConcat { it.changeInfo.operations.asFlow() }
+                .filter { it is TreeEditOpInfo || it is TreeStyleOpInfo }
+                .take(2)
+                .toList()
+        }
+
+        document.updateAsync {
+            it.tree().editByPath(listOf(0, 0, 0, 1), listOf(0, 0, 0, 1), TextNode("X"))
+            assertEquals(
+                "<doc><tc><p><tn>aXb</tn></p></tc></doc>",
+                it.tree().toXml(),
+            )
+
+            it.tree().styleByPath(listOf(0, 0, 0), mapOf("a" to "b"))
+        }.await()
+
+        assertContentEquals(
+            listOf(
+                TreeEditOpInfo(
+                    4,
+                    4,
+                    listOf(0, 0, 0, 1),
+                    listOf(0, 0, 0, 1),
+                    TreeNode("text", value = "X"),
+                    "$.t",
+                ),
+                TreeStyleOpInfo(
+                    6,
+                    7, // TODO: need to check if these are the actual right values???
```

Comment:
@skhugh 
`from`, `to` are correct for the given TC since it always indicates the closing tag.